File: src/test/java/ru/ezhov/rocket/action/configuration/ui/CreateRocketActionSettingsDialogTest.java
package ru.ezhov.rocket.action.configuration.ui;

import ru.ezhov.rocket.action.infrastructure.ReflectionRocketActionConfigurationRepository;
import ru.ezhov.rocket.action.infrastructure.ReflectionRocketActionUiRepository;

import javax.swing.JDialog;
import javax.swing.JFrame;
import javax.swing.SwingUtilities;

public class CreateRocketActionSettingsDialogTest {
    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            JDialog dialog = new JDialog();
            try {
                new CreateRocketActionSettingsDialog(
                        dialog,
                        new ReflectionRocketActionConfigurationRepository(),
                        new ReflectionRocketActionUiRepository()
                )
                        .show(rocketActionSettings -> {

                        });
            } catch (Exception e) {
                e.printStackTrace();
            }
        });
    }
}

File: src/test/kotlin/ru/ezhov/rocket/action/configuration/ui/CreateRocketActionSettingsDialogTest.kt
package ru.ezhov.rocket.action.configuration.ui

import ru.ezhov.rocket.action.configuration.infrastructure.ReflectionRocketActionConfigurationRepository
import ru.ezhov.rocket.action.infrastructure.ReflectionRocketActionUiRepository
import javax.swing.JFrame
import javax.swing.SwingUtilities

object CreateRocketActionSettingsDialogTest {
    @JvmStatic
    fun main(args: Array<String>) {
        SwingUtilities.invokeLater {
            val dialog = JFrame()
            dialog.setSize(1000, 900)
            val reflectionRocketActionConfigurationRepository = ReflectionRocketActionConfigurationRepository()
            val reflectionRocketActionUiRepository = ReflectionRocketActionUiRepository()
            try {
                CreateRocketActionSettingsDialog(
                    owner = dialog,
                    rocketActionConfigurationRepository = reflectionRocketActionConfigurationRepository,
                    rocketActionUiRepository = reflectionRocketActionUiRepository
                )
                    .show(object : CreatedRocketActionSettingsCallback {
                        override fun create(settings: TreeRocketActionSettings) {

                        }
                    })
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}